src/lib.rs

```diff
@@ -841,6 +840,11 @@ where
         self.index.iter().map(|x| x.len_items).sum()
     }
 
+    // Returns true if there are no values held by this reader.
+    pub fn is_empty(&self) -> bool {
+        self.index.iter().all(|x| x.len_items == 0)
+    }
```

Comment:
Note that these are actually going to be faster than the `.len() == 0` calls that they replace because they short-circuit on the first non-empty internal buffer.